pkg/tracing/tracing_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tracing_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/tracing"
+	"go.uber.org/zap"
+)
+
+func TestNewTracerProvider(t *testing.T) {
+	tp := tracing.New("test-serice")
+
+	tracer := tp.Tracer("tracer")
+	_, span := tracer.Start(context.TODO(), "example")
+
+	// tp.Tracer should return a nooptracer initially
+	// recording is always false for spans created by nooptracer
+	if span.IsRecording() {
+		t.Fatalf("Span is recording before configuration")
```

Comment:
super nits: context.background makes more sense than context.TODO IMO, since there's nothing todo, and I think t.Errorf makes more sense than Fatalf, since it's asserting test outputs rather than failing some setup steps

Reply:
Other tests are likely to fail if one of these fails, so it doesn't make much difference in reality.
But you are right that it is a better practice to use `t.Errorf` instead. I will make these nit changes along with any other feedbacks as part of the PR. Otherwise I will just add it to my next PR if that's okay.